### PR TITLE
Add/sync User.profile with profiles

### PR DESCRIPTION
**Pull request by Rotorsoft**
## Link to Issue
Closes: #8379

## Description of Changes
- Adds new field to model
- Migration with trigger to keep field in sync

## Test Plan
- Make sure changes in profiles are projected into new field

### CODE REVIEW

File: packages/commonwealth/server/migrations/20240708201453-add-profile-to-users.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                      'background_image', background_image
+                  ) AS profile_json
+            FROM "Profiles"
+            ORDER BY user_id, created_at
+        )
+        UPDATE "Users"
+        SET profile = COALESCE(FirstProfile.profile_json, '{}'::jsonb)
+        FROM FirstProfile
+        WHERE "Users".id = FirstProfile.user_id;
+        CREATE OR REPLACE FUNCTION update_user_profile() RETURNS TRIGGER AS $$
```
Comment (kurtisassad) [inline packages/commonwealth/server/migrations/20240708201453-add-profile-to-users.js:30]: Why do we have a trigger to update the user? Shouldn't it be done in application code for the user creation/updating? IMO having a trigger hides the logic.

Comment (Rotorsoft) [reply packages/commonwealth/server/migrations/20240708201453-add-profile-to-users.js:30]: This is just a temporary fix to keep the old and new models in sync... once we finish the bucket, the old model and trigger will be deleted

**Review by kurtisassad — APPROVED** [review]

File: packages/commonwealth/server/migrations/20240708201453-add-profile-to-users.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        BEGIN
+            IF TG_OP = 'INSERT' OR TG_OP = 'UPDATE' THEN
```
[the remark this answers is not in the document]
Comment (Rotorsoft) [reply packages/commonwealth/server/migrations/20240708201453-add-profile-to-users.js:32]: Don't think so, didn't find any.

File: packages/commonwealth/server/migrations/20240708201453-add-profile-to-users.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface) {
+    await queryInterface.sequelize.transaction(async (t) => {
+      await queryInterface.sequelize.query(
+        `
+        ALTER TABLE public."Users" ADD COLUMN "profile" JSONB NOT NULL DEFAULT '{}'::jsonb;
+        WITH FirstProfile AS (
+            SELECT DISTINCT ON (user_id)
```
[the remark this answers is not in the document]
Comment (Rotorsoft) [reply packages/commonwealth/server/migrations/20240708201453-add-profile-to-users.js:11]: Good point, removing

**Review by timolegros — APPROVED** [review]

**Pull request merged**